packages/plugin-ga-events-forwarder-browser/src/ga-events-forwarder.ts

```diff
@@ -0,0 +1,139 @@
+import { getGlobalScope } from '@amplitude/analytics-client-common';
+import { BaseEvent, BrowserClient, BrowserConfig, EnrichmentPlugin, Logger } from '@amplitude/analytics-types';
+import { GA_PAYLOAD_HOSTNAME_VALUE, GA_PAYLOAD_PATHNAME_VALUE } from './constants';
+import { isTrackingIdAllowed, isVersionSupported, parseGA4Events, transformToAmplitudeEvents } from './helpers';
+
+type BrowserEnrichmentPlugin = EnrichmentPlugin<BrowserClient, BrowserConfig>;
+interface Options {
+  trackingIds?: string | string[];
+}
+
+/**
+ * Returns an instance of `gaEventsForwarderPlugin`. Add this plugin to listen for events sent to Google Analytics,
+ * transform the events and send the events to Amplitude.
+ *
+ * ```html
+ * <script>
+ *   amplitude.add(gaEventsForwarder.plugin());
+ *   amplitude.init(<API_KEY>);
+ * </script>
+ * <script async src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX"></script>
+ * <script>
+ *   window.dataLayer = window.dataLayer || [];
+ *   function gtag(){
+ *     dataLayer.push(arguments);
+ *   }
+ *   gtag('js', new Date());
+ *   gtag('config', 'G-XXXXXXXXXX');
+ * </script>
+ * ```
+ *
+ * @param options An object containing plugin options. Options include `trackingIds`.
+ *
+ * @param options.trackingIds A Google Analytics tracking ID or a list of Google Analytics tracking IDs.
+ * This limits the plugin to only listen for events tracked with the specified tracking ID/s.
+ */
+export const gaEventsForwarderPlugin = ({ trackingIds = [] }: Options = {}): BrowserEnrichmentPlugin => {
+  const globalScope = getGlobalScope();
+  const trackingIdList = typeof trackingIds === 'string' ? [trackingIds] : trackingIds;
+  const isTrackingIdListValid = Array.isArray(trackingIdList) && !trackingIdList.some((id) => typeof id !== 'string');
+  let amplitude: BrowserClient | undefined = undefined;
+  let logger: Logger | undefined = undefined;
+  let preSetupEventQueue: BaseEvent[] = [];
+  let sendBeacon: undefined | ((url: string | URL, data?: BodyInit | null | undefined) => boolean);
+
+  /**
+   * Creates proxy for `navigator.sendBeacon` immediately to start listening for events.
+   * Google Analytics may start sending events before Amplitude SDK is initialized.
+   */
+  if (globalScope && isTrackingIdListValid) {
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    sendBeacon = globalScope.navigator.sendBeacon;
+
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    globalScope.navigator.sendBeacon = new Proxy(globalScope.navigator.sendBeacon, {
```

Comment:
I feel like I saw GA send events via fetch/xhr at least once, did you happen to test for that?

Reply:
confirmed that it's for UA and not GA4